pkg/speccheck/speccheck.go

```diff
@@ -110,6 +110,10 @@ func (sc *SpecCheck) compareActualVsSpec(spec, actual *unstructured.Unstructured
 	delete(updated.Object, "status")
 
 	if !equality.Semantic.DeepEqual(updated.Object, actualClone.Object) {
+		if (spec.GetKind() == "Secret") {
```

Comment:
This should be also checking the group of the object. Unique object type is identified by group+kind.
Versions evolve but they all represent the same thing withing group+kind e.g. you can read/write the same object using v1 and v1beta1.
Object cleanup works like that - it identifies "cleaner" functions by group+kind. See https://github.com/atlassian/smith/blob/master/pkg/cleanup/object_cleanup.go